astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -91,6 +81,7 @@ def cirs_to_icrs(cirs_coo, icrs_frame):
 
         # the distance in intermedrep is *not* a real distance as it does not
         # include the offset back to the SSB
+        srepr = cirs_coo.represent_as(SphericalRepresentation)
```

Comment:
Similarly, can avoid this by using `distance=cirs_coo.norm()` in the initializer below.